hash_practice/exercises.py

```diff
@@ -22,8 +51,76 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2) but I'm not sure. 
+        Space Complexity: O(1). While I create arrays, they're just temporary 

```

Comment:
Your answer is right, but your reasoning is wrong. It's right for the same reason as this generally takes O(1) time. We have a set grid size so it's always going to use up the same amount of time and the same amount of space.

However, in general, even if space is just temporarily used during calculation, that still counts as part of the space complexity. Most of the space that gets used up in these algorithms are temporary in that you no longer need them after the algorithm finishes, but they still take up the resources, so we have to account for them.